### PR TITLE
[FSTORE-838] asynchronous fg.insert/save

### PR DESCRIPTION
**Pull request by tdoehmen**
This PR changes the default for fg.insert()/save() to _not_ wait for the insert job to finish.

JIRA Issue: FSTORE-838

Priority for Review: -

Related PRs: -

**How Has This Been Tested?**

- [ ] Unit Tests
- [ ] Integration Tests
- [ ] Manual Tests on VM

**Checklist For The Assigned Reviewer:**

```
- [ ] Checked if merge conflicts with master exist
- [ ] Checked if stylechecks for Java and Python pass
- [ ] Checked if all docstrings were added and/or updated appropriately
- [ ] Ran spellcheck on docstring
- [ ] Checked if guides & concepts need to be updated
- [ ] Checked if naming conventions for parameters and variables were followed
- [ ] Checked if private methods are properly declared and used
- [ ] Checked if hard-to-understand areas of code are commented
- [ ] Checked if tests are effective
- [ ] Built and deployed changes on dev VM and tested manually
- [x] (Checked if all type annotations were added and/or updated appropriately)
```

### CODE REVIEW

File: python/hsfs/core/job.py [hunk 1]
```diff
@@ -104,3 +105,48 @@ def run(self, args: str = None, await_termination: bool = True):
             )
         )
         engine.get_instance().wait_for_job(self, await_termination=await_termination)
+
+    def get_state(self):
+        """Get the state of the job.
+
+        Returns the state of the job, which can be one of the following:
```
Comment (moritzmeister) [inline python/hsfs/core/job.py:112]: This formatting seems off.

File: python/hsfs/feature_group.py [hunk 2]
```diff
@@ -1647,7 +1647,7 @@ def save(
                   feature group.
                 * key `wait_for_job` and value `True` or `False` to configure
                   whether or not to the save call should return only
-                  after the Hopsworks Job has finished. By default it waits.
+                  after the Hopsworks Job has finished. By default it does not wait.
```
Comment (moritzmeister) [inline python/hsfs/feature_group.py:1650]: Can we add a Note admonition above to highlight from which version the default changed? or write it here.

Comment (moritzmeister) [reply python/hsfs/feature_group.py:1650]: I am confused I was expecting a new argument in save/insert? right now you are just changing the default? is this what you agreed on with @jimdowling

File: python/hsfs/engine/python.py [hunk 3]
```diff
@@ -946,15 +946,15 @@ def acked(err, msg):
             feature_group.backfill_job.run(
                 args=feature_group.backfill_job.config.get("defaultArgs", "")
                 + " -kafkaOffsetReset true",
-                await_termination=offline_write_options.get("wait_for_job", True),
+                await_termination=offline_write_options.get("wait_for_job", False),
```
Comment (bubriks) [inline python/hsfs/engine/python.py:949]: Shouldn't changes be also made here? https://github.com/logicalclocks/feature-store-api/pull/995/files#diff-e77e0390cdce0c2b0a51916239c6a232a59f58974c38edd465d1859fc5c53d48R76

Comment (tdoehmen) [reply python/hsfs/engine/python.py:949]: we don't want to wait for all jobs, just the hudi_fg.insert() jobs specifically.

File: python/hsfs/feature_group.py (hunk 2 above)
Comment (tdoehmen) [reply python/hsfs/feature_group.py:1650]: i discussed this solution with him

**Review by bubriks — APPROVED** [review]

File: python/hsfs/core/job.py (hunk 1 above)
Comment (tdoehmen) [reply python/hsfs/core/job.py]: pushed doc updates

File: docs/index.md [hunk 4]
```diff
@@ -124,7 +124,7 @@ td.save(feature_join)
 
 A short introduction to the Scala API:
 ```scala
-import com.logicalclocks.hsfs.spark._
+import com.logicalclocks.hsfs._
```
Comment (moritzmeister) [inline docs/index.md:127]: Why are you doing this, this was correct.

File: mkdocs.yml [hunk 5]
```diff
@@ -30,7 +30,7 @@ nav:
       - Query: generated/api/query_api.md
       - Transformation Functions: generated/api/transformation_functions_api.md
       - ValidationReport: generated/api/validation_report_api.md
-      - Job configuration: generated/api/job_configuration.md
+      - Job: generated/api/job.md
```
Comment (moritzmeister) [inline mkdocs.yml:33]: I don't understand why you created a new template but then don't use the old one anymore?

**Review by moritzmeister — CHANGES_REQUESTED** [review]

File: mkdocs.yml (hunk 5 above)
Comment (tdoehmen) [reply mkdocs.yml:33]: Because I included the job configuration in the job page now

File: docs/index.md (hunk 4 above)
Comment (tdoehmen) [reply docs/index.md]: Not sure why this happended, reverted it

File: mkdocs.yml (hunk 5 above)
Comment (moritzmeister) [reply mkdocs.yml:33]: ah okay there was no template for the job config.

**Review by moritzmeister — APPROVED** [review]

File: python/hsfs/feature_group.py (hunk 2 above)
Comment (tdoehmen) [reply python/hsfs/feature_group.py:1655]: Seems like we indeed wanted a new parameter, so I added a "wait"= parameter to fg.save/.insert() now

**Review by moritzmeister — APPROVED** [review]

**Pull request merged**